lib/condition/condition.ts

```diff
@@ -98,40 +99,30 @@ export class Range {
             case TREE.GreaterThanOrEqualToRelation:
                 this.min = value;
                 return;
-            case TREE.custom("InBetweenRelation"):
+            case TREE.custom("InBetweenRelation"): {
                 if (typeof value !== "string") {
                     throw (
                         "InBetweenRelation can only handle string values, not" +
                         typeof value
                     );
                 }
-                if (dataType === XSD.custom("gYear")) {
-                    const result = this.gYearToMinMax(value);
-                    if (!result) return;
-                    [this.min, this.max] = result;
-                    this.eqMin = true;
-                    this.eqMax = false;
-                } else if (dataType === XSD.custom("gYearMonth")) {
-                    const result = this.gYearMonthToMinMax(value);
-                    if (!result) return;
-                    [this.min, this.max] = result;
-                    this.eqMin = true;
-                    this.eqMax = false;
-                } else if (dataType === XSD.custom("date")) {
-                    const result = this.dateToMinMax(value);
-                    if (!result) return;
-                    [this.min, this.max] = result;
-                    this.eqMin = true;
-                    this.eqMax = false;
-                } else {
-                    // Check if it is a partial dateTime.
-                    const result = this.partialDateTimeToMinMax(value);
-                    if (!result) return;
-                    [this.min, this.max] = result;
-                    this.eqMin = true;
-                    this.eqMax = false;
+                const between = parseInBetweenRelation(
+                    value,
+                    dataType,
+                    this.defaultTimezone,
+                );
+                if (between) {
+                    if (this.min === undefined || between.min < this.min) {
+                        this.min = between.min;
+                        this.eqMin = true;
+                    }
+                    if (this.max === undefined || between.max < this.max) {
+                        this.max = between.max;
+                        this.eqMax = false;
+                    }
```

Comment:
This more in detail checking + assignments should happen in the `add` function below, instead of here, because in the constructor there will be no other bounds yet.